api/src/main/java/org/apache/iceberg/transforms/ProjectionUtil.java

```diff
@@ -254,4 +257,124 @@ private ProjectionUtil() {
     return predicate(predicate.op(), fieldName,
         Iterables.transform(predicate.asSetPredicate().literalSet(), transform::apply));
   }
+
+  static UnboundPredicate<Integer> fixInclusiveTimeProjection(UnboundPredicate<Integer> projected) {
+    if (projected == null) {
+      return projected;
+    }
+
+    // adjust the predicate for values that were 1 larger than the correct transformed value
```

Comment:
I think the mention of "correct" and "incorrect" in these comments might not be obvious to understand outside the context of this PR, as the reader has no prior knowledge of the issue we are fixing here from looking at the code base itself. Do we want to add some more explanation/link to the issue?

Reply:
I added docs to explain what "correct" and "incorrect" mean.